rosbag2_storage_default_plugins/src/rosbag2_storage_default_plugins/sqlite/sqlite_wrapper.cpp

```diff
@@ -37,7 +37,7 @@ SqliteWrapper::SqliteWrapper(
   if (io_flag == rosbag2_storage::storage_interfaces::IOFlag::READ_ONLY) {
     int rc = sqlite3_open_v2(
       uri.c_str(), &db_ptr,
-      SQLITE_OPEN_READONLY | SQLITE_OPEN_NOMUTEX, nullptr);
+      SQLITE_OPEN_READWRITE | SQLITE_OPEN_NOMUTEX, nullptr);
```

Comment:
Does this pose any danger of misuse? Would it be better to do a quick "open-close" to create the proper files, then still open as readonly?

Reply:
Good point.  I did a quick test, opening in RW, executing the first statement (`PRAGMA schema_version;`), closing it and reopening as RO works, and the *.db3-shm, *.db3-wal are created. I can implement it like that, but we're doubling the number of lines of code.

I'm wondering why it needs RW anyway. I'm not familiar with sqlite, so I don't understand how/why these WAL files are used. Maybe we are using it wrong?